### PR TITLE
feat: type nitro app runtime hooks

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

https://github.com/nuxt/nuxt/issues/21593#comment-2542384

### ❓ Type of change
- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This addresses a TODO comment to allow nitroApp hooks to be typed (and extended via module augmentation).

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/runtime/types.ts [hunk 1]
```diff
@@ -16,3 +19,11 @@ declare module "h3" {
     $fetch: typeof globalThis.fetch;
   }
 }
+
+export interface RuntimeNitroHooks {
+  "render:response": (
```
Comment (Hebilicious) [inline src/runtime/types.ts:24]: There's "render:html" too right ?

Comment (danielroe) [reply src/runtime/types.ts:24]: Quite right! But that is Nuxt-specific and so Nuxt would augment this interface with that hook, as well as the island render hook.

https://github.com/nuxt/nuxt/blob/71ef8bd3ff207fd51c2ca18d5a8c7140476780c7/packages/nuxt/src/core/runtime/nitro/renderer.ts#L228

**Review by pi0 — APPROVED** [review]
👍🏼

**Comment by danielroe** [comment]
Just FYI, I picked previous interface name to align with Nuxt's `RuntimeNuxtHooks`.

**Pull request merged**